lib/task/import/csvAuthorityRecordRelationImportTask.class.php

```diff
@@ -0,0 +1,185 @@
+    ]);
+
+    $this->namespace = 'csv';
+    $this->name = 'authority-relation-import';
+    $this->briefDescription = 'Import authority record relations using CSV data.';
+    $this->detailedDescription = <<<EOF
+      Import authority record relations using CSV data
+EOF;
+  }
+
+  /**
+   * @see sfTask
+   */
+  public function execute($arguments = [], $options = [])
+  {
+    parent::execute($arguments, $options);
+
+    $this->log('Importing relations...');
+    $this->import($arguments['filename'], $options['index']);
+    $this->log('Done.');
+  }
+
+  private function import($filepath, $indexDuringImport = false)
+  {
+    if (false === $fh = fopen($filepath, 'rb'))
+    {
+      throw new sfException('You must specify a valid filename');
+    }
+
+    // Load taxonomies into variables to avoid use of magic numbers
+    $termData = QubitFlatfileImport::loadTermsFromTaxonomies([
+      QubitTaxonomy::ACTOR_RELATION_TYPE_ID => 'actorRelationTypes',
+    ]);
+
+    $import = new QubitFlatfileImport([
+      'context' => sfContext::createInstance($this->configuration),
+
+      'status' => [
+        'actorRelationTypes' => $termData['actorRelationTypes'],
+        'actorIds'           => [],
+      ],
+
+      'variableColumns' => [
+        'sourceAuthorizedFormOfName',
+        'targetAuthorizedFormOfName',
+        'category',
+        'description',
+        'date',
+        'startDate',
+        'endDate'
+      ],
+
+      'saveLogic' => function($self)
+      {
+        // Figure out ID of the two actors
+        $sourceActor = QubitActor::getByAuthorizedFormOfName(
+          $self->rowStatusVars['sourceAuthorizedFormOfName'], ['culture' => $self->columnValue('culture')]);
+        $targetActor = QubitActor::getByAuthorizedFormOfName(
+          $self->rowStatusVars['targetAuthorizedFormOfName'], ['culture' => $self->columnValue('culture')]);
+
+        // Determine type ID of relationship type
+        $relationTypeId = array_search(
+          $self->rowStatusVars['category'],
+          $self->status['actorRelationTypes'][$self->columnValue('culture')]
+        );
+
+        if (!$relationTypeId)
+        {
+          throw new sfException(sprintf('Unknown relationship type %s:', $self->rowStatusVars['category']));
+        }
+        else
+        {
+          // Attempt to add relationship
+          if (empty($sourceActor) || empty($targetActor))
+          {
+            // Warn if actor is missing
+            $badActor = (empty($sourceActor))
+              ? $self->rowStatusVars['sourceAuthorizedFormOfName']
+              : $self->rowStatusVars['targetAuthorizedFormOfName'];
+
+            $error = sprintf('Actor "%s" does not exist', $badActor);
+            print $self->logError($error);
+          }
+          else
+          {
+            // Add relationship if it doesn't yet exist
+            if (!$this->checkIfRelationshipExists($sourceActor->id, $targetActor->id, $relationTypeId))
```

Comment:
Nitpick: imho `if (!$this->relationshipExists(...` reads better, and makes the method name shorter.